hardware_integration/include/isobus/hardware_integration/socket_can_interface.hpp

```diff
@@ -9,6 +9,7 @@
 //================================================================================================
 #ifndef SOCKET_CAN_INTERFACE_HPP
 #define SOCKET_CAN_INTERFACE_HPP
+#ifdef __linux__
```

Comment:
I also think we should use CMake for driver selection. However, using CMake with microcontrollers is often a hassle.
So a good approach would be to use PlatformioIO to handle source file selection as @GwnDaan already did before [this](https://github.com/GwnDaan/ISO11783-CAN-Stack/commit/1c880ec35deefef9fae7463dca57d837ce357ec3) commit.
I think selecting the source files in the library.json should be the preferred way and would make the #ifdef __linux__ obsolete. For handling other PlatformIO platforms, either ifdefs are needed or we could handle this in a Python script with the [extraScript](https://docs.platformio.org/en/latest/manifests/library-json/fields/build/extrascript.html) option.

Reply:
Yep, CMake with ESP-IDF does some unwanted behavior sometimes. For example pthread can't be found using `find_package(Threads REQUIRED)` in the main CMakeLists.txt while ptread certainly exists in esp-idf. I agree that selecting the source files in library.json is probably better. I'll change it back.

Reply:
> I think ideally we'd just select the different driver via CMake so only the needed one(s) get compiled and to avoid checking macro definitions. 

@ad3154 I attempted to implement this in 6c86f037c15ec1214b3445a039281e8c7f8138ba. Is it somewhat like what you had in mind?

Reply:
Exactly!

Reply:
Oh also, for what it's worth, I was able to compile for ESP32 by changing `find_package(Threads REQUIRED)` to just `find_package(Threads)`. The check then technically fails, but since ESP32 has threads and just doesn't implement this find_package step properly, it still works OK. Perhaps we should change it?

Reply:
We could also add a `if(NOT ESP_PLATFORM)` statement before `find_package(Threads REQUIRED)` to skip it completely? This way building for platforms without threads still fail successfully, and not confuse new users building for ESP thinking they don't have threading support.

Reply:
OK, that also sounds good to me!